packages/docregistry/src/default.ts

```diff
@@ -508,12 +515,12 @@ class MimeDocument extends Widget implements DocumentRegistry.IReadyWidget {
       data[this._mimeType] = model.toJSON();
     }
     let mimeModel = new MimeModel({ data });
-    if (!this._renderer) {
-      this._renderer = this.rendermime.createRenderer(this._mimeType);
-      (this.layout as BoxLayout).addWidget(this._renderer);
-      BoxLayout.setStretch(this._renderer, 1);
-    }
-    return this._renderer.renderModel(mimeModel);
+    return this._renderer.renderModel(mimeModel).then(() => {
+      // Handle a render after an activation.
+      if (this.node === document.activeElement) {
+        MessageLoop.sendMessage(this._renderer, Widget.Msg.ActivateRequest);
```

Comment:
I'm not sure about this logic. Shouldn't it be on the re-renderer to refocus itself if it had focus before re-rendering?

Reply:
The renderer might not have a focusable element at construction time, this allows it to defer taking focus in its `onActivateRequest` until after it has a focusable node.

Reply:
My point was more that this is subtle, and I'm not sure that the logic belongs in the render method.